tests/templates/kuttl/upgrade/test_nifi.py

```diff
@@ -37,30 +37,36 @@ def get_token(username, password, namespace):
                           help="Namespace the test is running in")
     args = vars(all_args.parse_args())
 
-    # shut up warnings
+    # disable warnings as we have specified non-verified https connections
     urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-    token = get_token(args['user'], args['password'], args['namespace'])
-
+    host = f"https://test-nifi-node-default-1.test-nifi-node-default.{args['namespace']}.svc.cluster.local:8443"
+    token = get_token(host, args['user'], args['password'])
     headers = {'Authorization': token}
-    url = 'https://test-nifi-node-default-1.test-nifi-node-default.' + args[
-        'namespace'] + '.svc.cluster.local:8443/nifi-api/controller/cluster'
-    cluster = requests.get(url, headers=headers, verify=False)  # , cert='/tmp/cacert.pem')
-
-    if cluster.ok:
-        cluster_data = json.loads(cluster.content.decode('utf-8'))
-    else:
-        print("Failed to get cluster data: ", cluster.status_code, " - ", cluster.content)
-
-    nodes = cluster_data['cluster']['nodes']
-
-    if len(nodes) != 3:
-        print("Cluster should have 3 nodes at this stage, but has: ", len(nodes))
-        exit(-1)
-
-    for node in nodes:
-        if node['status'] != "CONNECTED":
-            print('Node ', node['nodeId'], ' is in state ', node['status'], ' but should have been CONNECTED')
-            exit(-1)
 
-    print("Test succeeded!")
+    x = 0
+    while x < 15:
+        url = host + '/nifi-api/controller/cluster'
+        cluster = requests.get(url, headers=headers, verify=False)  # , cert='/tmp/cacert.pem')
+        if cluster.status_code != 200:
+            print("Waiting for cluster...")
+        else:
+            cluster_data = json.loads(cluster.content.decode('utf-8'))
+            nodes = cluster_data['cluster']['nodes']
+            if len(nodes) != 3:
+                print(f"Cluster should have 3 nodes at this stage, but has: {len(nodes)}")
```

Comment:
This checks for 3 or 2 nodes depending on the case. Does it make sense to make this another cli argument?

Reply:
Yes, added.